python/ray/autoscaler/autoscaler.py

```diff
@@ -512,9 +516,13 @@ def target_num_workers(self):
         ideal_num_nodes = int(np.ceil(cur_used / float(target_frac)))
         ideal_num_workers = ideal_num_nodes - 1  # subtract 1 for head node
 
+        initial_workers = self.config["initial_workers"]
+        aggressive = self.config["aggressive_autoscaling"]
         if self.bringup:
-            ideal_num_workers = max(ideal_num_workers,
-                                    self.config["initial_workers"])
+            ideal_num_workers = max(ideal_num_workers, initial_workers)
+        elif aggressive and ideal_num_workers >= 0:
+            # If we want any workers, we want at least initial_workers
+            ideal_num_workers = max(ideal_num_workers, initial_workers)
```

Comment:
Could we add a unit test for this mdoe?

Reply:
@ls-daniel are we planning to add a test for this in this pr? I would prefer to have a test added if that's possible.